### PR TITLE
Notify all in channel

### PR DESCRIPTION
**Pull request by adam000**
Use @all, @everyone, or @everybody to highlight the message for everyone in the channel.

### CODE REVIEW

File: nodejs/public/js/util.js [hunk 1]
```diff
@@ -32,7 +32,13 @@ Text: (function Text(){
 			var matches = text.match(/@([^\s]+)/g);
 			return matches && matches.some(function(match){
 				match = match.substr(1);
-				return match.length >= 2 && my_name.indexOf(match.toLowerCase()) >= 0;
+            // does it match @<name> or @all or @everyone or @everybody?
+            matches_name_or_all = 
+             my_name.indexOf(match.toLowerCase()) >= 0 ||
+             "all" == match.toLowerCase() ||
+             "everybody" == match.toLowerCase() ||
+             "everyone" == match.toLowerCase();
```
Comment (danielbeardsley) [inline nodejs/public/js/util.js]: ewww... backwards comparators!  Putting the constant value first when using binary operators is way unstandard and hard to read.

you should save the match.toLowerCase() and then compare them like this:

```
...
match == 'all' ||
match == ''everybody' ||
match == 'everyone';
```

**Comment by adam000** [comment]
http://cominor.com/trac/ticket/7242

File: nodejs/public/js/util.js (hunk 1 above)
Comment (adam000) [inline nodejs/public/js/util.js]: Fair enough, I'll fix that

File: nodejs/public/js/util.js [hunk 2]
```diff
@@ -31,8 +31,14 @@ Text: (function Text(){
 		mentionMatcher: function(text){
 			var matches = text.match(/@([^\s]+)/g);
 			return matches && matches.some(function(match){
-				match = match.substr(1);
-				return match.length >= 2 && my_name.indexOf(match.toLowerCase()) >= 0;
+				match = match.substr(1).toLowerCase();
+            // does it match @<name> or @all or @everyone or @everybody?
+            matches_name_or_all =
```
Comment (danielbeardsley) [inline nodejs/public/js/util.js]: One more :-(

you should put a "var" in front of this variable, otherwise it will end up being global.

Comment (adam000) [inline nodejs/public/js/util.js]: Fixed! Silly mistake ;)

File: nodejs/public/js/util.js [hunk 3]
```diff
@@ -31,8 +31,14 @@ Text: (function Text(){
 		mentionMatcher: function(text){
 			var matches = text.match(/@([^\s]+)/g);
 			return matches && matches.some(function(match){
-				match = match.substr(1);
-				return match.length >= 2 && my_name.indexOf(match.toLowerCase()) >= 0;
+				match = match.substr(1).toLowerCase();
```
Comment (xiongchiamiov) [inline nodejs/public/js/util.js:34]: This line's got tabs. ;)

Comment (adam000) [inline nodejs/public/js/util.js:34]: The file has inconsistent tabbing. I'll fix this in a future commit. (along with the rest of the repository, as necessary)

**Pull request merged**